be/src/storage/vectorized/convert_helper.cpp

```diff
@@ -592,6 +593,40 @@ class StringToOtherTypeConverter : public TypeConverter {
     }
 };
 
+// Convert string to json
+// JSON needs dynamic memory allocation, which could not fit in TypeInfo::from_string
+template <>
+class StringToOtherTypeConverter<OLAP_FIELD_TYPE_JSON> : public TypeConverter {
+public:
+    using CppType = typename CppTypeTraits<OLAP_FIELD_TYPE_JSON>::CppType;
+
+    StringToOtherTypeConverter() = default;
+    ~StringToOtherTypeConverter() = default;
+
+    Status convert(void* dst, const void* src, MemPool* memPool) const override {
+        return Status::InternalError("not supported");
+    }
+
+    Status convert_datum(TypeInfo* src_typeinfo, const Datum& src, TypeInfo* dst_typeinfo, Datum& dst,
+                         MemPool* mem_pool) const override {
+        Slice source;
+        if (src.is_null()) {
+            source = "null";
+        } else {
+            source = src.get_slice();
+        }
+        JsonValue json;
+        RETURN_IF_ERROR(JsonValue::parse(source, &json));
+
+        // TODO(mofei) optimize this, avoid clone
+        StatusOr<JsonValue*> copy = json.clone(mem_pool);
```

Comment:
It seems too trick in JsonValue::clone, Is it possible add TYPE JsonValue in Datum::variant

Reply:
Well, TYPE_JSON in Datum::Variant is `JsonValue*`, which is a pointer type, not a class.

Why pointer type? Because class type may create a lot of copy in computation, which is not suitable for vectorized engine.

But pointer type indeed introduce a lot of problem, for example need be specialized in storage engine, and predicate.

Fundamentally, type system needs to be designed for both scalar type and complex user-defined types, the later could not be handled well currently.

Reply:
I meant to add a new TYPE in Datum::Variant

Reply:
I see. 

But I'm afraid that it could not solve all problems. Value stored in datum is based on `TYPE_JSON` and `RunTimeTypeTraits`. Even if the datum could store a class type, nobody could leverage it.

Reply:
Finally, to avoid hack mempool, store an `JsonValue` in datum as `convert_datum` result. 

In this way, JSON in datum could be both `JsonValue` and `JsonValue*`, and `JsonColumn::append_datum` could accept either of them.